proto/api/v1/api.proto

```diff
@@ -799,3 +800,61 @@ option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
     }
   }
 };
+
+service PluginService {
+  rpc ListPlugins(ListPluginsRequest) returns (ListPluginsResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugins"
+      response_body: "plugins"
+    };
+  };
+  rpc GetPlugin(GetPluginRequest) returns (GetPluginResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugin/{name}"
+      response_body: "plugin"
+    };
+  };
+}
+
+message ListPluginsRequest {
+}
+
+message ListPluginsResponse{
+  repeated PluginSpecifications plugins = 1;
+}
+
+message GetPluginRequest {
+  string name = 1;
+}
+
+message GetPluginResponse {
+  PluginSpecifications plugin = 1;
+}
+
+message PluginSpecifications{
+  enum Type {
```

Comment:
I would move this `Type` inside of the `Parameter` message, to make it clear that this is a parameter type.